src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -420,6 +421,10 @@ private void HandleChangesForStaticIndexes(DatabaseRecord record, long index, Li
 
                 try
                 {
+                    if (definition.Name.StartsWith(Constants.Documents.Indexing.SideBySideIndexNamePrefix, StringComparison.OrdinalIgnoreCase))
+                    {
+                        throw new InvalidOperationException($"Index name cannot start with {Constants.Documents.Indexing.SideBySideIndexNamePrefix} but got {definition.Name} of index {name}");
```

Comment:
would want to double-confirm what will happen here?

Reply:
The index definitions here are the record's, the database record shouldn't contain SideBySide indexes cause they are being created only locally in the node, not in the cluster, so definition of SideBySide index shouldn't be passed to this method.

Reply:
so what will happen if we will hit this? server will crash? dataabase will not load?

Reply:
The server will crash, but it shouldn't happen from the first place.

Reply:
I can switch it to 'Debug.Assert' check if you want.

Reply:
The concern is that someone could already have the database record affected by this issue. Then after the upgrade to a newer version with this fix included they could encounter problems. If this means that server will crash during the startup then it's not an option to add this check. Although I'm surprised that this will cause the crash. How can that be? From the brief look at the code it appeared that that worst case it will create a faulty index (the replacement one).

Reply:
So I'll remove this check?

Reply:
It depends. If that really causes the crash then definitely we cannot have such check. Although I'd like to understand how this can cause the crash because I thought that the only "inconvenience" would be that the index becomes faulty - which would be acceptable since it's about unintentional replacement index. 

It's potentially risky change so we need to have good understanding what are possible consequences. So my question is still open - how can this cause a crash of server?

Reply:
It doesn't cause the crash, it only turns the index into faulty

Reply:
Sorry I missed your comment somehow. Okay, that's is fine then.